Vignettes/using_cchsflow.Rmd

```diff
@@ -0,0 +1,286 @@
+
+```{r eval= FALSE}
+install.packages("devtools")
+library(devtools)
+install_github("Big-Life-Lab/cchsflow")
+```
+
+```{r results= 'hide', message = FALSE, warning=FALSE}
+library(cchsflow)
+```
+
+We use the `bind_rows` function in the [dplyr](https://www.rdocumentation.org/packages/dplyr/versions/0.7.8) package to combine datasets.
+
+```{r results = 'hide', message = FALSE, warning =FALSE}
+library(dplyr)
+```
+
+For this vignette, we use the `kable` function in the [knitr](https://yihui.name/knitr/) package to illustrate transformation.
+```{r results = 'hide', message = FALSE, warning =FALSE}
+library(knitr)
+library(kableExtra)
+```
+
+## Steps
+
+### Step 1. Load the CCHS datasets
+
+Load the CCHS datasets as well as the two variable worksheets. For illustration, we will create mock CCHS datasets for the years 2001 and 2013. 
+
+```{r}
+cchs_mock2001 <- data.frame(DHHA_SEX = c(2, 1, 1, 6), DHHAGAGE = c(3, 4, 6, 6), FVCADTOT = c(12, 4, 20, 9))
+cchs_mock2013 <- data.frame(DHH_SEX = c(1, 2, 1), DHHGAGE = c(2, 1, 1), FVCDTOT = c(25, 15, 6))
+```
+
+Did you notice that the names for the variables are slightly different in the two mock databases? That isn't a mistake: in the 2001 CCHS the variable for sex is `DHHA_SEX` and in 2013 CCHS the variable is `DHH_SEX`. 
+
+Don't worry, `cchsflow` is here to help! `variable_details.csv` contains the rules to harmonize those two variables into a common variable name. In the CCHS, the categories for `sex` are consistient: 1 = males, 2 = females. `variable_details.csv` provides instructions for how to harmonizecategory values or labels, if they change across survey cycles. Variable labes are discussed in later vignettes.
+
+```{r, echo=FALSE, message=FALSE, warning=FALSE}
+library(DT)
+datatable(variables, options = list(pageLength = 5))
+cat("`variables.csv` includes", length(unique(variables$variable)), "variables that can be transformed. The variables are divided into", length(unique(variables$section)), "sections, and", length(unique(variables$subject)), "subjects.")
+```
+
+### Step 2. Transform variables
+
+To call the `rec_with_table()` function to transform the variables of a CCHS dataset:
+
+1. Create a name for your transformed dataset (e.g. cchsYear1)
+2. Indicate the database of the CCHS dataset in the `data_source` argument
+3. Indicate the name of `variable_details.csv` in the `variable_details` argument
+4. Indicate the name of the CCHS dataset (this can be found in the databaseStart section of `variable_details.csv`).
+
+Your R command when calling the function should look like this:
+
+`cchsTransformedYear1 <- rec_with_table(data_source = cchsYear1, variable_details = variable_details, dataset_name = "cchsYear1")`
```

Comment:
this might cause confusion why not use the samples we generated

Reply:
Was trying to write pseudocode for this